test/provider_os_memory_multiple_numa_nodes.cpp

```diff
@@ -111,3 +136,102 @@ TEST_P(testNumaNodes, checkNumaNodesAllocations) {
     int retrieved_numa_node_number = retrieve_numa_node_number(ptr);
     ASSERT_EQ(retrieved_numa_node_number, numa_node_number);
 }
+
+// Test for allocations on numa nodes with mode preferred. It runs for all available
+// numa nodes obtained from the get_available_numa_nodes_numbers() function.
+TEST_P(testNumaNodes, checkModePreferred) {
+    int numa_node_number = GetParam();
+    umf_os_memory_provider_params_t os_memory_provider_params =
+        UMF_OS_MEMORY_PROVIDER_PARAMS_TEST;
+    os_memory_provider_params.maxnode = numa_node_number + 1;
+    numa_bitmask_setbit(nodemask, numa_node_number);
+    os_memory_provider_params.nodemask = nodemask->maskp;
+    os_memory_provider_params.numa_mode = UMF_NUMA_MODE_PREFERRED;
+    initOsProvider(os_memory_provider_params);
+
+    umf_result_t umf_result;
+    umf_result =
+        umfMemoryProviderAlloc(os_memory_provider, alloc_size, 0, &ptr);
```

Comment:
you doing here only single small allocation, but please note that each of this modes differs in OOM scenario - you should test for that

Reply:
I was thinking of adding all the OOM tests in the next PR.